src/helpers.rs

```diff
@@ -339,4 +339,29 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         Ok(())
     }
+
+    /// Sets the last error variable
+    fn set_last_error(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+        let errno_place = this.machine.last_error.unwrap();
+        this.write_scalar(scalar, errno_place.into())
+    }
+
+    /// Gets the last error variable
+    fn get_last_error(&mut self) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+        let errno_place = this.machine.last_error.unwrap();
+        this.read_scalar(errno_place.into())?.not_undef()
+    }
+
+    /// Sets the last error variable using a `std::io::Error`. It fails if the error cannot be
+    /// transformed to a raw os error succesfully
+    fn set_last_error_from_io_error(&mut self, e: std::io::Error) -> InterpResult<'tcx> {
```

Comment:
`set_last_error_from_io` would be slightly shorter and still "informative enough", I feel.

Reply:
Oh and while you are at it, `fs.rs` also has a "consume" method that I find similarly confusing. Any proposals for what else it could be called?

Reply:
Sure, sounds reasonable.

In fact, is there any reason that's in `fs.rs` and not here?

Reply:
Whatever you prefer.